### DISCUSSION TITLE
hardware per automazione VMC

### DISCUSSION THREAD

**Discussion by mtaliesin** [indomus/forum#2571] [Hardware & Firmware]
Buongiorno a tutti,
per favore vorrei un vostro consiglio su come automatizzare la mia VMC. 
La foto mostra i collegamenti del controller che, in pratica, serve solo a scegliere la velocità della ventola da 0 (spenta) a 3;
qualora servisse, allego anche il mauale.
Per favore, che componente mi consigliate per controllarlo con Home assistant? Mi va bene che possa comunicare sia tramite wifi che zigbee.
Grazie
[asset-istruzioni-per-linstallazione-zehnder-comfosense.pdf](https://github.com/indomus/forum/files/12269038/asset-istruzioni-per-linstallazione-zehnder-comfosense.pdf
![1691311945263](https://github.com/indomus/forum/assets/125162112/6f7761b1-6aa9-4663-b1a1-6a492a12a74b)
zione-zehnder-comfosense.pdf)

**Comment by zanac** [comment]
Io ho una vecchia CA350 che dentro ha una bella RS232 e con questa diventa tutto facile.... se hai questa ti do dritte! Se hai una nuova Q350 invece non so aiutarti.
Ho anche creato una card custom molto carina:
![image](https://github.com/indomus/forum/assets/21194919/45d52a18-e109-47cf-92da-7fedc093dd25)

**Reply by mtaliesin** [reply]
Bellissima la card!
La mia VMC non ha la rs232,ma grazie lo stesso per la risposta

**Reply by Peeka49** [reply]
Bellissima questa card, per caso riesci a condividerla?

**Comment by zanac** [comment]
che modello hai cmq per curiosità? se non è la 350 vecchia ma la Q350 nuova basta autocostruirsi una piccola basetta esp32 in realtà sai!
https://github.com/vekexasia/comfoair-esp32

A questo punto le mie card dato che si basano su integrazioni sono usabili... persino quella della mappa termica che prende i dati dal sensore della temperatura esterna:
![image](https://github.com/indomus/forum/assets/21194919/01c07f82-adea-49bd-8843-5c25575dbf52)

**Reply by mtaliesin** [reply]
La mia è una comfoair 200. 
Sono completamente a zero in elettronica; anche io ho pensato che una esp32 potesse essere utile, ma non ho idea di come vada collegata al controller  di cui ho messo la foto.

**Comment by zanac** [comment]
Ma la 200 ha una RS232 come la mia...
https://www.phstore.co.uk/zehnder-comfoair-200

La RS232 è in realtà mascherata tramite una RJ45, ho tagliato un cavo di rete, collegato all'interno seguenod il datasheet (di seguito quello della tua), e da qui tirato fuori RX e TX della 232!!!! Sei fuori strada sai... lo schema ESP32 è completamente cannato nel tuo caso ed è tutto più facile.
https://22c.lt/atsisiusti/4031

**Reply by mtaliesin** [reply]
Grazie per le info.
Adesso devo solo scoprire dove accedere a quel connettore.
Purtroppo mi hanno montato la VMC in un posto infame : a più di 4 metri d'altezza su una rampa di scale. Già cambiare i filtri è una mezza impresa.

Alla prima occasione, vedo se il connettore è accessibile.
Casomai ti disturberò di nuovo per quella card.
Per il momento, ti ringrazio di nuovo

**Comment by zanac** [comment]
Guarda, il datsshet è chiaro, devi smontare cmq tutta la parte anteriore della vmc, poi io all'epoca ho messo un cavo di rete, tagliato l'altra estremità, mi sono segnato i colori del connettore che ho inserito dentro, fatto un foro sulla vmc per portare fiori il cavo..... Poi con calma ho collegato il tutto su un adattatore cinese che virtualizza rs232 in tcp

**Reply by mtaliesin** [reply]
Ok, grazie : in effetti non sembra troppo difficile per le mie scarse capacità.
Come ti ho detto, il problema principale è accedere alla macchina, ma mi sto attrezzando, anche per cambiare i filtri.
Ne approfitto per un'altra domanda : alla fine, con l'adattatore cinese cui hai accennato a cosa è collegato? Trasmette in wifi? Perché non ho un raspberry o simili da mettere lì sopra e sarebbe difficile portargli la corrente e/o cavo di rete 
Grazie ancora

**Comment by zanac** [comment]
L'adattatore è ethernet.... Unico dubbio è che il protocollo della 200 sia lo stesso della 350 ma penso di sì.

**Reply by mtaliesin** [reply]
Peccato... 
Dovrò fare un po' di lavori prima di poterci provare.
Grazie ancora per i consigli e le informazioni

**Comment by zanac** [comment]
No fermo, rileggi bene, l'adattatore serve solo per virtualizzare la seriale... te la provo dire in altre parole: una volta che hai il cavo che esce dalla RS232 il primo step è collegare il cavo fisicamente ad un PC (tramite ad esempio una RS232/USB) con un PC con home assistant (andando con un notbook volante vicino alla VMC) e provare che il protocollo della 350 sia lo stesso della 200. Una volta che hai appurato che funziona allora l'adattatore ha il solo scopo di virtualizzare su ethernet la RS232 in modo da non dover mettere il pc con home assistant vicino alla VMC perchè a questo punto può essere fisicamente distante. Così ti è più chiaro?

**Reply by mtaliesin** [reply]
Ok, credo di aver capito un po' meglio 
Per piacere mi daresti un Link dell'adattatore, così ci dò un'occhiata?

**Comment by zanac** [comment]
https://it.aliexpress.com/item/32860667371.html

Io uso questo e funziona alla grande

**Reply by mtaliesin** [reply]
Perfetto, grazie mille

**Comment by mtaliesin** [comment]
Ciao, scusa se ti disturbo di nuovo... Per piacere, volevo un paio di chiarimenti:
come collego l'adattatore che mi hai linkato https://it.aliexpress.com/item/32860667371.html al raspberry con home assistant? tramite cavo lan? ho visto che ne esiste uno wifi: potrebbe andar bene lo stesso?
Ultima domanda: su home assistant che integrazione uso? Come faccio a farlo riconoscere?
Grazie mille

**Comment by zanac** [comment]
1. L'adattatore ethernet va collegato alla ethernet di casa
2. La versione wifi non funziona bene da test fatti da altri. Eventualmente puoi portare la LAN tramite un powerline mal che vada.
3. Nessuna integrazione, scheduli allo startup di Linux un servizio che tira su dei sensori MQTT che saranno discoverati in automatico... questo servizio lo trovi su https://github.com/adorobis/hacomfoairmqtt/tree/master, una volta attivo troverai automaticamente i sensori della VMC, l'unico dubbio come ti dicevo è che non l'abbiamo mai testata con il tuo modello ma solo con la 350 e la 550.
4. ribadisco che prima di fare qualsiasi esperimento con la configurazione effettiva (che quindi nel tuo caso sarebbe adattatore cinese+powerline) farei delle prove di comunicazione portando un PC portatile vicino alla VMC e collegandolo magari con una semplice RS232/USB.

**Reply by MrGio80** [reply]
Ciao Zanac,
Grazie per le informazioni che hai condiviso.
Ho anche io una Comfoair 200.
Mi è tutto chiaro per la parte elettrica.

Mi perdo sul punto 3.
Come installo questa parte su Raspberry?
Che interfaccia devo configurare per fargli leggere i dati seriali su ethernet?

Puoi darmi qualche dettaglio?

Grazie in anticipo!

**Reply by zanac** [reply]
ok, ti mando un po' di schermate, perchè non è banale...

step 1: ho acquistato l'interfaccia cinese che avevo linkato, quindi l'ho configurata per avere un IP statico (192.168.2.8) e li l'ho settata come da screenshot che segue:
![image](https://github.com/indomus/forum/assets/21194919/f4ae6412-c8c4-429a-bde9-ff389b6588ac)

step 2: sul PC che gestisce home assistant (non ho rasbperry ma un PC) ho installato "socat" e creato uno script come da screenshot che espone una seriale virtuale partendo dal server su 192.168.2.8
![image](https://github.com/indomus/forum/assets/21194919/456bcd24-50c0-4f07-8c64-206da475eb7c)

step 3: scaricato il software ca350.py da questo link: https://github.com/adorobis/hacomfoairmqtt/tree/master/src con relativo config

step 4: adattato il config di default come da screenshot per puntarlo alla seriale virtuale creata da socat e al server MQTT di home assistant

![image](https://github.com/indomus/forum/assets/21194919/3a4672ae-ea87-43cb-8812-b65df05f63e0)

**Reply by MrGio80** [reply]
Grazie!
Socat non penso sia possibile usarlo su Raspberry con l'immagine di Home Assistant purtroppo.

Provo a usare un convertitore usb rs232.

Temo dovrò approfittare ancora del tuo supporto...

Buona giornata

**Reply by zanac** [reply]
Secondo me invece basta che lo installi tramite docker... però qui ti rimando a qualcuno che ha esperienza con docker e potacci vari....

https://github.com/peterforeman/homeassistant-socat

**Comment by mtaliesin** [comment]
Grazie mille: velocissimo e preciso. 
I prossimi giorni mi arrampico sulla vmc e vedo se riesco a passare un cavo di rete... 

-----Original Message-----

**Reply by zanac** [reply]
Ma, scusa se insisto, ma prima di passare cavi hai aperto la VMC e guardato se ha l'attacco? una volta trovato l'attacco non devi passare il cavo di rete effettivo ma infilare il cavo di rete in quell'attacco, tagliare l'altra estremità del cavo di rete e saldarci una RS232 secondo il pinout del datasheet... poi lasciando questo cavo penzolini comodamente con un notebook volante e un cavo RS232-USB fai esperimenti di comunicazione.

**Comment by mtaliesin** [comment]
Non l' ho ancora fatto : è tra i controlli che devo fare appena mi arrampico. 
Grazie 


-----Original Message-----

**Comment by mtaliesin** [comment]
Ciao. Finalmente ho aperto la VMC ed ho dato un'occhiata. Da quel che vedo, non ho bisogno di passare da rj45 a RS232, ce n'è una libera sul fianco della macchina (seconda foto)
Se non ho capito male le tue istruzioni, dovrei
- prendere un cavo RS232 - usb, collegarlo al  PC per vedere se tutto funziona;
- prendere un cavo  RS232 M - RS232 M e collegarlo all'adattore ethernet;
giusto?
Grazie dell'aiuto e della pazienza 
![rs232](https://github.com/indomus/forum/assets/125162112/8aa6e6d4-d941-4f34-ac2c-9dff7df2d9e8)
![1693725912109](https://github.com/indomus/forum/assets/125162112/0ef07475-ede1-42a1-a727-7c0541cd09b5)
![schema.jpg](https://github.com/indomus/forum/assets/125162112/602d51f1-4a8a-41ad-8b74-90fb663b0484)

**Reply by zanac** [reply]
no sbagliato.
Attieniti alle indicazioni date non inventarti cose pensando di semplificare.
Quella porta RS232 che citi usa un altro protocollo a livello software e non funziona per il nostro scopo.
Devi essere tedesco e seguire le indicazioni non fare di testa tua.

**Reply by mtaliesin** [reply]
ok, grazie

**Comment by zanac** [comment]
ecco uno schema del pinout da preparare... ovviamente in base se scegli un connettore maschio o femmina devi seguire il disegno corretto (al massimo prova invertirli se hai dubbi, io mi ero preparato all'epoca un cavo con morsetti apposta per fare prove prima di saldare il tutto)

rj45 pin 2 -> rs232 pin 3 - yellow
rj45 pin 3 -> rs232 pin 2 - white
rj45 pin 8 -> rs232 pin 5 - green

![image](https://github.com/indomus/forum/assets/21194919/e352d7dd-ee12-4604-ab32-bc4fc689dd8d)
![image](https://github.com/indomus/forum/assets/21194919/873efabe-9d5f-4bca-a4c7-69bf5f943f1a)

**Comment by mtaliesin** [comment]
Grazie. Appena mi arrivano i pezzi mi metto al lavoro 

-----Original Message-----

**Comment by skydado** [comment]
@zanac
Card bellissima: complimenti!!
Mi inserisco in questo thread perchè anche io ho una VMC modello Aspirnova RC250 il cui controllo remoto ha le seguenti caratteristiche di comunicazione:

Protocollo: MODBUS RTU
 Comunicazione: RS485 port – (115200 / 19200 / 9600)-8-N-1 (Half- Duplex)
 Controllo errore: CRC16 (polinomio 0xA001)
 Fine messaggio: considera pacchetto MODBUS completo, dopo 5 ms
dall’ultimo byte ricevuto
 Formato messaggio:
Master -> Slave:
INDIRIZZO SLAVE (1byte) – CODICE FUNZIONE (1byte) – DATI (Nbytes) – CRC16 (2bytes) Slave -> MASTER:
INDIRIZZO SLAVE (1byte) – CODICE FUNZIONE (1byte) – DATI(Nbytes) – CRC16 (2bytes)

Secondo la tua esperienza, sarebbe smartizzabile seguendo più o meno gli stessi passi?
Nel caso apro nuovo thread.

**Comment by zanac** [comment]
La card in se è utilizzabile in quanto "generica", dovresti però riscriverti il server di backend che estrae e pubblica i dati via MQTT... di fatto quindi la risposta è no purtroppo :(
